packages/libp2p/src/dcutr/dcutr.ts

```diff
@@ -214,9 +253,67 @@ export class DefaultDCUtRService implements Startable {
     }
   }
 
+  /**
+   * This function encapsulates the Simultaneous Connect step (#5) of the [DCUtR protocol](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md#the-protocol)
+   * @param relayedConnection
+   * @param multiaddrs
+   * @param signal
+   */
+  async handleSimultaneousConnect (relayedConnection: Connection, multiaddrs: Multiaddr[], rtt: number, signal: AbortSignal): Promise<void> {
+    logB('waiting for half RTT')
+    // ..and starts a timer for half the RTT measured from the time between
+    // sending the initial Connect and receiving the response
+    // TODO: Find a way to simulate network latency for tests, because rtt is always <= 4ms
+    await delay(rtt / 2)
+
+    const tcpMultiaddrs = multiaddrs.filter(ma => ma.protoNames().includes('tcp'))
+    const quicMultiaddrs = multiaddrs.filter(ma => ma.protoNames().includes('quic'))
```

Comment:
`quic` needs to be `quic` or `quic-v1`.

It'll also cause any webtransport addresses to be selected which will then cause `Promise.any` to throw due to `simultaneousConnectQuic` throwing.

Reply:
You can also use matchers from [@multiformats/multiaddr-matcher](https://multiformats.github.io/js-multiaddr-matcher/modules.html) for this.

Reply:
I was looking for those matchers and decided to just put this for now as a placeholder. thanks for the callout

Reply:
https://github.com/libp2p/js-libp2p/pull/1932/commits/9bc8680c54f3eddd53807b0cd5a39d0160507d53